File: Angula2BaseStructure/Common/Entities/Dorm/Room.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Entities.Students;


namespace Entities.Dorm
{
    public class Room
    {
        public int RoomId { get; set; }
        public int Name { get; set; }
        public int TotalPlaces { get; set; }
        public int AvaliablePlaces { get; set; }
        public virtual List<Student> StudentsInRoom { get; set; }
    }
}

File: Angula2BaseStructure/Angula2BaseStructure/Scripts/App/Entities/Dorm/Room.ts
import {Student} from '../Student/Student';
export class Room {
    roomId: number;
    name: string;
    totalPlaces: number;
    avaliablePlaces: number;
    studentsInRoom: Array<Student>;

    constructor() {
        this.studentsInRoom = new Array<Student>();
    }
}